src/webview/memory-webview-view.tsx

```diff
@@ -65,6 +70,7 @@ class App extends React.Component<{}, MemoryAppState> {
         columnContributionService.register(new AsciiColumn());
         decorationService.register(variableDecorator);
         this.state = {
+            messageParticipant: { type: 'webview', webviewId: '' },
```

Comment:
I'd probably that we make this field nullable, as the default value isn't actually meaningful.

Reply:
I also thought about that. The question is then how do we handle the potential `undefined` value?
This would mean the resulting global context object would conform to this interface.
```export interface WebviewMenuContext {
    messageParticipant?: WebviewIdMessageParticipant,
    visibleColumns: string,
    showRadixPrefix: boolean,
}
```ts

So in theory, in a command handler for a context menu command we would  always have to check whether `ctx.messageParticipant` is defined before sending a message to the webview. In practice this check is unnecessary, because the participant is only undefined on initial render and should be correctly set once the initial settings are transferred from the extension.

At some point we have to make the implicit assumption that the `messageParticipant` is set to the correct/valid value.
If would rather not have this on the user facing side (i.e. command context)

Reply:
Fair enough. It's really just an artifact of the interaction between React and the webview lifecycle. I suppose the question is what it looks like to a user if the synchronization goes bad for some reason. Is it a no-op in the VSCode messenger package, or does it throw an error; if error, how transparent is it? Could we provide a better message if something did go wrong?

Reply:
I quickly tested this. Using an invalid webview will be handeled by vscode-messenger silently (or at least not with a direct user-facing error message). A simple error message is logged in the debug console but that's it.

The `messageParticipant` is part of the inital settings notification that is dispached as soon as the webview is ready.
So in practice we can only end up with an invalid (emtpy id)  `messageParticipant` if
- The webview never sends the ready notification
- or something goes wrong when retrieving the display settings from the vscode configuration.

Both cases are very unlikely, and if they should occur we probably have more pressing concerns than the invalid `messageParticipant` as the entire webview-extension communication flow is not working as expected.

If we want to consider these corner cases we should probably handle this on a more generic level e.g. freezing the webview and/or display a loading indicator until the initial settings are received. In addition, we could add timeout error handling for cases where the extension never receives the `webviewReady` notification. But this should probably be tackled in a follow-up PR.

Reply:
Failing silently in unlikely circumstances is good enough for me. I was worried that it would throw an error, and if VSCode encounters an unhandled error during command execution, it'll show a very ugly dialog. As long as we avoid that, I'm happy.